.github/workflows/sync-api-vmlinux-cron.yaml

```diff
@@ -0,0 +1,115 @@
+name: Sync api.h and vmlinux.h from latest kernel
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 1 * *' # first day of each month
+
+jobs:
+  sync-api-vmlinux:
+    runs-on: ${{ (matrix.arch == 'arm64' && 'ubuntu-24.04-arm') || 'ubuntu-24.04' }}
+    strategy:
+      matrix:
+        arch: [ x86, arm64 ]
+        include:
+          - arch: x86
+            tumbleweed_repo: 'https://download.opensuse.org/tumbleweed/repo/oss/x86_64/'
+          - arch: arm64
+            tumbleweed_repo: 'https://download.opensuse.org/ports/aarch64/tumbleweed/repo/oss/aarch64/'
+    steps:
+      - name: Install bpftool
+        uses: mtardy/setup-bpftool@9bc044304197616a6321d452e04cc059e006d8e2 # v1.0.4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download most recent kernel package from opensuse tumbleweed
+        run: |
+          image=$(wget -qO- ${{ matrix.tumbleweed_repo }} | grep "kernel-default-base-*" | head -n1 | awk '{split($0,file,"\""); print file[4]}')
+          wget -O image.rpm ${{ matrix.tumbleweed_repo }}$image
+          mkdir image && cd image
+          rpm2cpio ../image.rpm | cpio -idmv
+          prefix="./kernel-default-base-"
+          suffix=".rpm"
+          version=${image#"$prefix"}
+          version=${version%"$suffix"}
+          echo "KERNEL_VERSION=$version" >> $GITHUB_ENV
+
+      - name: Download extract-kernel script from upstream linux repo
+        run: |
+          wget -O extract-vmlinux https://raw.githubusercontent.com/torvalds/linux/refs/heads/master/scripts/extract-vmlinux
+          chmod +x extract-vmlinux
+
+      - name: Extract vmlinuz
+        run: |
+          ./extract-vmlinux image/usr/lib/modules/*/vmlinux.xz > decomp-vmlinuz
+
+      - name: Generate updated vmlinux_generated
+        run: |
+          mkdir out
+          bpftool btf dump file decomp-vmlinuz format c > out/vmlinux_generated_${{ matrix.arch }}.h
+          # Add an autogenerated comment
+          sed -i '1i// Autogenerated from opensuse tumbleweed ${{ env.KERNEL_VERSION }} kernel' out/vmlinux_generated_${{ matrix.arch }}.h
+          sed -i 's/__VMLINUX_H__/__VMLINUX_GENERATED_H__/g' out/vmlinux_generated_${{ matrix.arch }}.h
+
+      - name: Sync bpf_helper_defs.h from upstream libbpf repo (x86 only)
+        if: matrix.arch == 'x86'
+        env:
+          PATCH_LONG_INT: 'map_update_elem;map_delete_elem;probe_read;perf_event_output;probe_read_str;probe_read_user;probe_read_kernel'
```

Comment:
On older kernels using new signature for these helpers causes troubles with the verifier.
This env variable is later used to patch each of the element to return `int` instead of `long`.